recipes/mongo-c-driver/all/conanfile.py

```diff
@@ -89,6 +85,8 @@ def validate(self):
             raise ConanInvalidConfiguration("with_ssl=windows only allowed on Windows")
         if self.options.with_sasl == "sspi" and self.settings.os != "Windows":
             raise ConanInvalidConfiguration("with_sasl=sspi only allowed on Windows")
+        if tools.Version(self.version) >= "1.21.0" and self.settings.os == "Windows" and not self.options.shared:
+            raise ConanInvalidConfiguration("shared build doesn't allow on Windows after 1.21.0")
```

Comment:
Could you elaborate please? Does it fail to build because it's officially unsupported upstream since 1.21.0? Or because there is some sort of error which could be fixed easily? Windows or only Visual Studio? This ConanInvalidConfiguration is quite problematic, it means that it raises for default options on Windows.